src/plugins/dashboard/public/application/embeddable/grid/_dashboard_grid.scss

```diff
@@ -41,6 +41,16 @@
  * .dshLayout-withoutMargins only affects the panel styles themselves, see ../panel
  */
 
+/**
+ * 1. Fixes fullscreen mode for saved searches. Otherwise, sibling grids can obscure the grid in fullscreen mode.
+ * "euiDataGrid__restrictBody" is set to body element when a grid is in fullscreen
+ * https://github.com/elastic/kibana/issues/134032#issuecomment-1178091867
+ * It can be removed if we don't use inline styles (z-index, position) for Dashboard panels (react-grid-item)
+ */
+.euiDataGrid__restrictBody .dshDashboardGrid__item {
+  position: static !important; /* 1 */
```

Comment:
@constancecchen Your suggestion works great for desktop view, thanks! 

But there is something weird with mobile view. Although it has `position: static !important` https://github.com/elastic/kibana/blob/fe0e70258fd18198b3b8fa98776a4d9b538545a6/src/plugins/dashboard/public/application/embeddable/grid/_dashboard_grid.scss#L125 the issue with overlapping grids is still present. What is different for mobile view in grids implementation?